### ISSUE TITLE
ETag caching not working

### ISSUE THREAD

**Issue by tbhmens** [tinyhttp/tinyhttp#270]
**Describe the bug**

**With cache enabled in the browser:**
![with browser-side cache enabled](https://user-images.githubusercontent.com/57283066/122643977-fd0c7280-d112-11eb-864b-3b411c01f27b.png)
**Without cache enabled in the browser:**
![without browser-side cache enabled](https://user-images.githubusercontent.com/57283066/122644040-5d031900-d113-11eb-881e-10c2a8594aee.png)
Both send a 200 and the data.

**To Reproduce**

1. Initialize App with freshnessTesting set to true  
```js
import { App } from "@tinyhttp/app";
new App({
    settings: {
        freshnessTesting: true
    }
})
    .get("/", (_, res) => res.send("Things"))
    .listen(8080, () => console.log("Listening on port 8080"));
```
2. Open localhost:8080 in the browser
3. Refresh
4. Both requests sent got HTTP code 200, even though they both have the same ETag

**Expected behavior**

tinyhttp sends a 304 not modified code on the refresh, so the browser gets it from it's cache.

**Versions**

* `node`: 16.3.0
* `@tinyhttp/app`: 1.3.8

**Comment by talentlessguy** [comment]
thanks for finding this bug! I actually forgot to put a freshness check in res.send... sorry

will be fixed in a moment

**Comment by talentlessguy** [comment]
should be fixed in tinyhttp@1.3.9, please check

**Comment by tbhmens** [comment]
![image](https://user-images.githubusercontent.com/57283066/122655526-39f85980-d153-11eb-91db-4b98f50cbeff.png)
It now sends 304 even without a If-None-Match, though the first time I sent the request it did send a 200.

**Comment by tbhmens** [comment]
Found the problem! on line 49 req.fresh is a function and not a boolean, and functions are truthy.
https://github.com/tinyhttp/tinyhttp/blob/65a618524a51961f2e2555909c8fddfe4793e57e/packages/send/src/send.ts#L49

**Comment by talentlessguy** [comment]
damn it I'm sorry for my idiocy, tomorrow I'll fix that

**Comment by talentlessguy** [comment]
@tbhmens could you help on this one please?

here I set the property but now in tests it simply doesn't set 304 even if header is matched: https://github.com/tinyhttp/tinyhttp/blob/bbaddcaadddc009a49715d49f4939d0c3ade7d4f/packages/app/src/extend.ts#L96

the tests fail:

https://github.com/tinyhttp/tinyhttp/blob/bbaddcaadddc009a49715d49f4939d0c3ade7d4f/tests/core/app.test.ts#L918-L930

https://github.com/tinyhttp/tinyhttp/blob/bbaddcaadddc009a49715d49f4939d0c3ade7d4f/tests/core/request.test.ts#L135-L148

https://github.com/tinyhttp/tinyhttp/blob/bbaddcaadddc009a49715d49f4939d0c3ade7d4f/tests/modules/send.test.ts#L96-L114

**Comment by tbhmens** [comment]
commented on your commit, i think it doesnt work since in extend.ts the result etag header isnt set yet, so it's checking the req etag to an undefined. since the result etag _can_ only be set in send, i think send will just have to call fresh as a function instead

**Comment by talentlessguy** [comment]
@tbhmens thanks for looking into the problem, the thing is that in Express `fresh` is a property... so I can't make it a function. Do you know any better solution than converting it to a function?

**Comment by tbhmens** [comment]
Possibly, when setting the etag header, also updating the fresh value using getFreshOrStale? A sort of hook into the setHeader function that checks if the ETag header is getting set, and if so, updating fresh

**Comment by tbhmens** [comment]
Alternatively, use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

**Comment by talentlessguy** [comment]
@tbhmens could you please pull request this? :P

**Comment by tbhmens** [comment]
Sure! #272 👍

**Comment by talentlessguy** [comment]
@tbhmens published in `1.3.10`, please check

also, sorry for noise with IssueHunt, the bot bugged cuz it wasn't installed in an org
